src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -181,32 +182,32 @@ class OrderedRVD(
     require(newPartitioner.kType isPrefixOf typ.kType)
 
     new OrderedRVD(
-      typ.copy(
-        partitionKey = typ.key.take(newPartitioner.partitionKey.getOrElse(newPartitioner.kType.size)),
-        key = typ.key.take(newPartitioner.kType.size)),
+      typ.copy(key = typ.key.take(newPartitioner.kType.size)),
       newPartitioner,
       ContextRDD(new RepartitionedOrderedRDD2(this, newPartitioner.rangeBounds)))
   }
 
   def localSort(newKey: Array[String]): OrderedRVD = {
     require(newKey startsWith typ.key)
     require(newKey.forall(typ.rowType.fieldNames.contains))
-    require(partitioner.partitionKey.isDefined)
+    require(partitioner.satisfiesAllowedOverlap(typ.key.length - 1))
```

Comment:
Why is this key.length - 1?

Reply:
Allowed overlap `key.length - 1` means that keys in separate partitions may be equal in at most the first `key.length - 1` fields. So this is the weakest precondition which disallows having the same key occur in multiple partitions. We need this here because we need to be able to collect all rows sharing a key together to sort by some additional new key fields. If the rows sharing a key were split across partitions, we couldn't do the sorting locally.